runtime/integration-tests/src/runtime_apis/pools.rs

```diff
@@ -0,0 +1,187 @@
+				borrower.clone(),
+				Role::PoolRole(PoolRole::Borrower),
+			);
+
+			let token_name = BoundedVec::try_from("SuperToken".as_bytes().to_owned())
+				.expect("Can't create BoundedVec");
+			let token_symbol =
+				BoundedVec::try_from("ST".as_bytes().to_owned()).expect("Can't create BoundedVec");
+
+			// Setting up metadata for the pool currency
+			<development_runtime::OrmlAssetRegistry as Mutate>::register_asset(
+				Some(CurrencyId::AUSD.into()),
+				orml_asset_registry::AssetMetadata {
+					decimals: 18,
+					name: token_name.to_vec(),
+					symbol: token_symbol.to_vec(),
+					existential_deposit: 0_u128.into(),
+					location: None,
+					additional: CustomMetadata::default(),
+				},
+			)
+			.expect("Registering asset metadata should not fail");
+
+			// Creating a pool
+			<development_runtime::PoolSystem as PoolMutate<AccountId, PoolId>>::create(
+				admin.clone(),
+				admin.clone(),
+				pool_id.clone(),
+				vec![
+					TrancheInput {
+						tranche_type: TrancheType::Residual,
+						seniority: None,
+						metadata: TrancheMetadata {
+							token_name,
+							token_symbol,
+						},
+					},
+					TrancheInput {
+						tranche_type: TrancheType::NonResidual {
+							interest_rate_per_sec: Rate::one(),
+							min_risk_buffer: Perquintill::from_percent(10),
+						},
+						seniority: None,
+						metadata: TrancheMetadata {
+							token_name: BoundedVec::default(),
+							token_symbol: BoundedVec::default(),
+						},
+					},
+				],
+				CurrencyId::AUSD,
+				10_000 * 10_000_000_000,
+				None,
+			)
+			.expect("Pool creation should not fail");
+
+			//
+			// Initalising a pool and creating a loan
+			//
+			let loan_nft_class_id: CollectionId = 1_u64.into();
+			<development_runtime::Uniques as Create<AccountId>>::create_collection(
+				&loan_nft_class_id,
+				&admin.clone(),
+				&admin.clone(),
+			)
+			.expect("collection creation should not fail");
+
+			<development_runtime::Loans>::initialise_pool(
+				RawOrigin::Signed(admin).into(),
+				pool_id,
+				loan_nft_class_id,
+			)
+			.expect("initialisation of pool should not fail");
+
+			let collateral_class: CollectionId = 2_u64.into();
+			<development_runtime::Uniques as Create<AccountId>>::create_collection(
+				&collateral_class,
+				&borrower.clone(),
+				&borrower.clone(),
+			)
+			.expect("class creation should not fail");
+
+			let instance_id = LoanId::from(1_u32);
+
+			<development_runtime::Uniques as NMutate<AccountId>>::mint_into(
+				&collateral_class.into(),
+				&instance_id.into(),
+				&borrower,
+			)
+			.expect("mint should not fail");
+
+			let collateral = Asset(collateral_class, instance_id);
+
+			<development_runtime::Loans>::create(
+				RuntimeOrigin::signed(borrower),
+				pool_id,
+				collateral,
+			)
+			.expect("Creation of loan should not fail");
+		})
+		.with_api(|api, latest| {
+			let valuation = api.portfolio_valuation(&latest, 3).unwrap();
+			assert_eq!(valuation, Some(0));
```

Comment:
Would be nice if we could warp the time a little and use a fixed value instead.

Reply:
I was curious about that too. Can you point me to the fudge feature where we can warp time?

Reply:
you can use 

```
// Not sure if milliseconds or seconds here...
pallet_timestamp::Now::<Runtime>::set(TIME);
```

Reply:
But in the `startup` closure